src/ophyd_async/panda/_table.py

```diff
@@ -54,23 +56,23 @@ class SeqTableRow:
 
 
 class SeqTable(TypedDict):
-    repeats: npt.NDArray[np.uint16]
-    trigger: Sequence[SeqTrigger]
-    position: npt.NDArray[np.int32]
-    time1: npt.NDArray[np.uint32]
-    outa1: npt.NDArray[np.bool_]
-    outb1: npt.NDArray[np.bool_]
-    outc1: npt.NDArray[np.bool_]
-    outd1: npt.NDArray[np.bool_]
-    oute1: npt.NDArray[np.bool_]
-    outf1: npt.NDArray[np.bool_]
-    time2: npt.NDArray[np.uint32]
-    outa2: npt.NDArray[np.bool_]
-    outb2: npt.NDArray[np.bool_]
-    outc2: npt.NDArray[np.bool_]
-    outd2: npt.NDArray[np.bool_]
-    oute2: npt.NDArray[np.bool_]
-    outf2: npt.NDArray[np.bool_]
+    repeats: NotRequired[pnd.Np1DArrayUint16]
+    trigger: NotRequired[Sequence[SeqTrigger]]
+    position: NotRequired[pnd.Np1DArrayInt32]
+    time1: NotRequired[pnd.Np1DArrayUint32]
+    outa1: NotRequired[pnd.Np1DArrayBool]
+    outb1: NotRequired[pnd.Np1DArrayBool]
+    outc1: NotRequired[pnd.Np1DArrayBool]
+    outd1: NotRequired[pnd.Np1DArrayBool]
+    oute1: NotRequired[pnd.Np1DArrayBool]
+    outf1: NotRequired[pnd.Np1DArrayBool]
+    time2: NotRequired[pnd.Np1DArrayUint32]
+    outa2: NotRequired[pnd.Np1DArrayBool]
+    outb2: NotRequired[pnd.Np1DArrayBool]
+    outc2: NotRequired[pnd.Np1DArrayBool]
+    outd2: NotRequired[pnd.Np1DArrayBool]
+    oute2: NotRequired[pnd.Np1DArrayBool]
+    outf2: NotRequired[pnd.Np1DArrayBool]
```

Comment:
I'm not sure why this needs to change. We have 2 use cases for array fields:
- A `TypedDict` with fields `npt.NDArray[dtype]` that corresponds to a PVA table
- A `BaseModel` with fields that are numpy arrays according to how pydantic-numpy wants to define them

I thought this was the first of those, so I'm confused as to why we have pydantic-numpy things in it? Did I miss something?

Reply:
This allows for validation in `SeqTableInfo`:

https://github.com/bluesky/ophyd-async/blob/a8ad9bf27f41e468f22a1c4e2c915d9a67197bca/src/ophyd_async/panda/_trigger.py#L16-L19

Pydantic needs the type annotation from `pydantic_numpy` in the `SeqTable` in order to validate on it. It doesn't seem to check the exact type of the numpy array (`bool`, `uint32`, etc.) but it will validate based on whether it's a numpy array.

`NotRequired` is added since a plan may only pack `SeqTable` with a subset of rows relevant to the experiment.

Reply:
As discussed on slack, I think we should put this back to how it was, then disable strict checking of the `sequence_table` field above until #442 is solved